src/Engine/ProtoCore/RuntimeCore.cs

```diff
@@ -188,6 +189,9 @@ public void RegisterDllTypes(List<System.Type> dllTypes)
 
         // Cached at levels for the current call.
         public List<AtLevel> AtLevels;
+
+        // GUIDs of executed ASTs.
+        public HashSet<Guid> ExecutedAstGuids; 
```

Comment:
The whole idea for my recommendation of creating `RecordExtecutedGraphNode` was because `ExecutedAstGuids` can be made `private` and only publicly readable. Sorry for not making it clear!